src/activities/calendar/calendar.js

```diff
@@ -0,0 +1,141 @@
+var minRange //sum of min. visible month and year on calendar for navigation bar next/prev button visibility.
+var maxRange //sum of max. visible month and year on calendar for navigation bar next/prev button visibility.
+var correctAnswer
+var isCorrectAnswer
+var mode
+
+function start(items_, dataset_) {
+    items = items_
+    dataset = dataset_.get()
+    numberOfLevel = dataset.length
+    currentLevel = 0
+    initLevel();
+}
+
+function stop() {
+}
+
+function initLevel() {
+    currentSubLevel = 1;
+    items.bar.level = currentLevel + 1
+    currentLevelConfig = dataset[currentLevel][0][0]
+    setCalendarConfigurations()
+    initQuestion();
+}
+
+function nextLevel() {
+    if(numberOfLevel <= ++currentLevel) {
+        currentLevel = 0
+    }
+    initLevel();
+}
+
+function previousLevel() {
+    if(--currentLevel < 0) {
+        currentLevel = numberOfLevel - 1
+    }
+    initLevel();
+}
+
+// Configure calendar properties for every level.
+function setCalendarConfigurations() {
+    minRange = Number(currentLevelConfig["minimumDate"].slice(5,7)) + Number(currentLevelConfig["minimumDate"].slice(0,4)) - 1;
+    maxRange = Number(currentLevelConfig["maximumDate"].slice(5,7)) + Number(currentLevelConfig["maximumDate"].slice(0,4)) - 1;
+    mode = currentLevelConfig["mode"]
+    items.calendar.navigationBarVisible = currentLevelConfig["navigationBarVisible"]
+    items.calendar.minimumDate = currentLevelConfig["minimumDate"]
+    items.calendar.maximumDate = currentLevelConfig["maximumDate"]
+    items.calendar.visibleYear = currentLevelConfig["visibleYear"]
+    yearSelected = currentLevelConfig["visibleYear"]
+    items.calendar.visibleMonth = currentLevelConfig["visibleMonth"]
+    monthSelected = currentLevelConfig["visibleMonth"]
+    items.answerChoices.visible = (mode === "findDayOfWeek") ? true : false
+    items.okButton.visible = !items.answerChoices.visible
+    currentDataSet = dataset[currentLevel][1]
+    currentDataSet = Core.shuffle(currentDataSet)
+    items.score.numberOfSubLevels = currentDataSet.length
+    items.score.currentSubLevel = currentSubLevel
+}
+
+function initQuestion() {
+    if(currentDataSet.length < currentSubLevel) {
+        items.bonus.good("lion")
+    }
+    else {
+        items.score.currentSubLevel = currentSubLevel
+        items.questionItem.text = currentDataSet[currentSubLevel-1]["question"]
+        correctAnswer = currentDataSet[currentSubLevel-1]["answer"]
+    }
+}
+
+function updateScore() {
+    if(isCorrectAnswer) {
+        items.questionDelay.start()
+        items.okButtonParticles.burst(20)
+        items.score.playWinAnimation()
+        currentSubLevel++;
+    }
+    else
+        items.bonus.bad("lion")
+    isCorrectAnswer = false
+}
+
+function checkAnswer() {
+    // For levels having questions based on day of week only.
+    if(mode === "findDayOfWeek") {
+        if(dayOfWeekSelected === correctAnswer["dayOfWeek"]) {
+            isCorrectAnswer = true
```

Comment:
it would be better to have a local variable instead of a global one and pass it as argument in the function